File: 118-pascals-triangle/pascals-triangle.java
class Solution {
    public int nCr(int n,int r)
    {
        int prod =1;
        int re = (n-r)<r?(n-r):r;
        for(int i=0;i<re;i++)
        {
            prod = prod*(n-i);
            prod = prod/(i+1);
        }
        return prod;
    }
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();
        for(int i=0;i<numRows;i++)
        {
            List<Integer> l = new ArrayList<>();
            if(i==0)
            {
                l.add(1);
                res.add(l);
            }
            else
            {
            for(int j=0;j<=i;j++)
            {
                l.add(nCr(i,j));
            }
            res.add(l);
            }
            
        }
        return res;
    }
}

File: 118-pascals-triangle/pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        res =[]
        def gen(n,res):
            if n ==1:
                res.append([1])
                return [1]
            x = [0] + gen(n-1,res) + [0]
            li=[]
            for i in range(1,len(x)):
                s = x[i-1]+x[i]
                li.append(s)
            res.append(li)
            return li
        row = gen(numRows,res)
        return res